src/ray/core_worker/pubsub/publisher.cc

```diff
@@ -216,10 +230,23 @@ bool Publisher::UnregisterSubscriber(const NodeID &subscriber_node_id) {
   return erased;
 }
 
-bool Publisher::UnregisterSubscription(const NodeID &subscriber_node_id,
-                                       const ObjectID &object_id) {
+void Publisher::CheckDeadSubscribers() {
   absl::MutexLock lock(&mutex_);
-  return subscription_index_.EraseEntry(object_id, subscriber_node_id);
+  for (const auto &it : subscribers_) {
+    const auto &subscriber_id = it.first;
```

Comment:
Nit: Same comment as below, I'd usually prefer that `subscriber_id` is defined (or just using `it.first` directly) within the `if (disconnect)` scope since it's only used when unregistering a subscriber.